pinot-tools/src/main/java/org/apache/pinot/tools/service/PinotServiceManager.java

```diff
@@ -199,6 +199,7 @@ public boolean stopPinotInstance(ServiceStartable instance) {
       ServiceRole role = instance.getServiceRole();
       String instanceId = instance.getInstanceId();
       LOGGER.info("Trying to stop Pinot [{}] Instance [{}] ...", role, instanceId);
+      ServiceStatus.setServiceStatusCallback(instanceId, new ServiceStatus.ShuttingDownServiceStatusCallback());
       instance.stop();
```

Comment:
I could use some help understanding the impact of doing this. On our end (and for most people) I assume this is desired so load balancers actually stop sending queries to these machines while they're going down. But will this affect pinot internally somehow? Should this be a new configuration? Should it be limited to brokers?

Reply:
hmm. in this case, shall we just make set each pinot component(controller/broker/server) has a field `_isShuttingDown` and set them to true here?
Then all existing component healthcheck will take that field to return status.